File: Code/AudioPlayersDemos/Android-Media-Player/app/src/main/java/com/istudio/player/handlers/PlayerHandler.java
package com.istudio.player.handlers;

import android.os.Handler;
import android.os.Message;
import android.os.RemoteException;
import android.widget.Switch;

import androidx.annotation.NonNull;

import com.istudio.player.service.PlayerService;

/**
 * We need the reference to the player service inside this handler
 *
 */
public class PlayerHandler extends Handler {

    private PlayerService mPlayerService;

    public PlayerHandler(PlayerService playerService){
        this.mPlayerService = playerService;
    }

    @Override
    public void handleMessage(@NonNull Message msg) {
        // Here we shall perform different operations based on the arguments
        switch(msg.arg1){
            case 0: // Play
                mPlayerService.play();
                break;
            case 1: // Pause
                mPlayerService.pause();
                break;
            case 2: // isPlaying
                // If player is playing return "1" else return "0"
                int isPlaying = mPlayerService.isPlayerPlaying() ? 1 : 0 ;
                Message message = Message.obtain();
                message.arg1 = isPlaying;
                if(msg.arg2==1){
                    message.arg2 = 1;
                }
                message.replyTo = mPlayerService.getMMessenger();
                // Send the message to the activity
                try {
                    msg.replyTo.send(message);
                } catch (RemoteException e) {
                    throw new RuntimeException(e);
                }
                break;
        }

    }



}

File: Code/AudioPlayersDemos/Android-Media-Player/app/src/main/java/com/istudio/player/handlers/PlayerHandler.kt
package com.istudio.player.handlers

import android.os.Handler
import android.os.Message
import com.istudio.player.service.PlayerService

class PlayerHandler(private val mPlayerService: PlayerService) : Handler() {

    override fun handleMessage(msg: Message) {
        handleInCommingMessages(msg)
    }

    private fun handleInCommingMessages(msg: Message) {
        when (msg.arg1) {
            0 -> mPlayerService.play()
            1 -> mPlayerService.pause()
            2 -> {

                val isPlaying = if (mPlayerService.isPlayerPlaying()) 1 else 0
                // Prepare the message
                val message = Message.obtain()
                message.arg1 = isPlaying
               if(msg.arg2==1){
                   message.arg1 = 1
               }
                message.replyTo = mPlayerService.mMessenger
                try {
                    // msg.replyTo gives the messenger attached to this message
                    msg.replyTo.send(message)
                } catch (exception: Exception) {
                    exception.printStackTrace()
                }
            }
        }
    }

}
